app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,13 @@
+class FoodsController < ApplicationController
+  before_action :user_search
+  
+  def index
+    @search = FoodFacade.get_foods_with(params[:q])
+  end
+
+  private
+
+  def user_search
+    @user_search = params[:q]
+  end
+end
```

Comment:
This will definitely work. If you keep it this way, I'd use `@user_search` instead of `params[:q]` on line 5. I don't really have a preference for or against it. The way you currently have it set up, it would make just as much sense to have line 11 in the index action.. But, what this could allow for you to do is do some error handling before it hits the index action. For example, in the before action, you could make sure that `q` comes through, and if it didn't, flashing a message and rendering the root path again.